### ISSUE TITLE
Ag-Grid not working in IE11 and React,redux

### ISSUE THREAD

**Issue by ranjeethasachin** [ag-grid/ag-grid#1971]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
When Including AG-GRID latest version 13 in a REact-Redux based app, the GRid loading fails with a Null Reference.  Line No 67 in AGCheckBox.js            this.eChecked.appendChild(utils_1.Utils.createIconNoSpan('checkboxChecked', this.gridOptionsWrapper, null, svgFactory.createCheckboxCheckedIcon));
Shows below error 
"Unable to get property 'appendChild' of undefined or null reference"
I do not have any checkbox in my application.

The Previous Version which we were using was Version 10. GRid was perfectly loading in it.

**Expected behavior**
Ag-Gird should load seamlessly in IE11 using React,Redux

**Minimal reproduction of the problem with instructions**
A simple Redux app with a Store.
index.js Code
render(
	<Provider store={store}>
		<div className=<My APP>>
			
			<App/>
		</div>
	</Provider>,
	document.getElementById('app')
);

Add.js Code
render() {
       

        return (
            <div style={{height:"100%"}}>
                <AppViewTemp/>
                
            </div>
        );
    }
AppViewTemp.js code

render() {
        return (
            <div style={{height:"100%"}}>
                
                <MyGRidTemp/>
            </div>
        )
    }

MyGRidTemp.js code

    render() {
        var coldefs = [
    {headerName: "Make", field: "make"},
    {headerName: "Model", field: "model"},
    {headerName: "Price", field: "price"}
];
var rowData1 = [
    {make: "Toyota", model: "Celica", price: 35000},
    {make: "Ford", model: "Mondeo", price: 32000},
    {make: "Porsche", model: "Boxter", price: 72000}
];
alert(coldefs);
alert(rowData1);

        return (
            <div style={{ height: "100%" }}>
              <AgGridReact
    columnDefs={coldefs}
    rowData={rowData1}>
</AgGridReact>  
            </div>
        )
    }

}

**Please tell us about your environment:**
Local Windows OS, Webpack dev server 1.12.0, React 15, redux 3, react-redux 5,

* **ag-Grid version:** X.X.X
13.3.0

* **Browser:**
IE11

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
ES6

**Comment by petyosi** [comment]
Hi,

Please follow the issue template when raising new issues. Specifically, please provide a minimal reproduction of the issue that we can use as a baseline to investigate. We will reopen the issue once we have that. Thank you.

**Comment by anatolyg** [comment]
@petyosi I am seeing this issue as well. It seems that the bean is wired and initialized before the elements are available in the DOM. Is there a way to disable loading of ag-checkbox bean altogether?

**Comment by petyosi** [comment]
@anatolyg - Can you provide a repro of that? We will examine it further. Thank you.

**Comment by anatolyg** [comment]
@petyosi I resolved the issue by polyfilling the `constructor.name` property in IE. The real issue is that the grid uses `constructor.name` to track metadata about the various beans, and in IE, that metadata is grouped under `undefined`.

**Comment by bradsandifer** [comment]
Anyone find a resolution to this problem?

**Comment by EdAllonby** [comment]
Hi @anatolyg, do you have a snippet of your polyfill you could share? I'm encountering what seems to be the same error as the original, but haven't found a suitable polyfill yet.

**Comment by anatolyg** [comment]
@EdAllonby yep, here it is. This is from this bug: https://github.com/ag-grid/ag-grid/issues/1815#issuecomment-324881863

```
(function(Object, Function) {
    //* IE10 environment feature upgrade
    // i don't know who the hell on ag-grid side thought it would be a good idea to rely on Object.prototype.__proto__
    // __proto__ has ALWAYS been a very very VERY VERY VERY.... BAD CODE STYLE!
    // the DEV implemented that should be fired INSTANTLY because he/she has no idea of javascript (and especially javascript performance) at all!
    // not only the official documentations show BIG RED WARNINGS of using __proto__:
    //   https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/proto
    // there are better supported alternatives like Object.prototype.getPrototypeOf:
    //   https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Object/getPrototypeOf
    //
    // i know my hack is a very dirty modification to the IE<=10 environment...
    // thank you ag-grid for that shit and all the side effects coming out of it!
    //
    // same applies to Function.prototype.name  (which is not even supported on IE>10 - did you completely drop IE support???)
    //
    var propdef;

    // Object.prototype.__proto__
    if(!('__proto__' in Object.prototype)) {
      propdef = {
        configurable: false,
        enumerable: false,
        get: function() {
          //console.log("called getter for Object.prototype.__proto__");
          return Object.getPrototypeOf(this);
        }
      };
      if(typeof Object.setPrototypeOf !== "undefined") {
        propdef.writable = true;
        propdef.set = function(value) {
          return Object.setPrototypeOf(this, value);
        };
      }
      Object.defineProperty(Object.prototype, '__proto__', propdef);
    }

    // Function.prototype.name
    if(!('name' in Function.prototype)) {
      propdef = {
        configurable: false,
        enumerable: false,
        get: function() {
          var fn = this;
          //console.log("called getter for Function.prototype.name");
          if(typeof fn.__name !== "undefined")
            return ;
          var match = ("" + fn).match(/^\s*function\s+([^\(\s]*)\s*/);
          fn.__name = match ? match[1] : "";
          return fn.__name;
        }
      };
      Object.defineProperty(Function.prototype, 'name', propdef);
    }

  })(window.Object, window.Function);
```

**Comment by EdAllonby** [comment]
Thanks for the response @anatolyg, much appreciated, We'll give this another try.

**Comment by JabbyPanda** [comment]
From my testing I conclude that this particular error with `ag-grid` happens only if `babel-polyfill` https://babeljs.io/docs/usage/polyfill/#usage-in-browser is loaded in the IE browser

**Comment by JabbyPanda** [comment]
https://jsfiddle.net/Jabbypanda/ea4angwm/1/show/ here is example of the issue originally reported

This JSFiddle loads just load ag-grid 17.1 and babel-polyfil 6.26 from CDN

**Comment by JabbyPanda** [comment]
what is also important to mention is that `babel-polyfill` library has to be loaded **AFTER** ag-grid to reproduce this issue

![image](https://user-images.githubusercontent.com/255913/39925812-69e1e26a-552d-11e8-8f5b-db98723c125b.png)

If this library is loaded *BEFORE* ag-grid module, then the error mentioned in original bug report does not occur

**Comment by vlcr** [comment]
**JabbyPanda,**

> what is also important to mention is that babel-polyfill library has to be loaded **AFTER** ag-grid to reproduce this issue

`Babel` should be included **before** all your code. Please take a look at [the docs](https://babeljs.io/docs/usage/polyfill/).

> If this library is loaded **BEFORE** ag-grid module, then the error mentioned in original bug report does not occur

As you would expect it to, as that's the right configuration for it.

**Comment by MatthewHerbst** [comment]
Thanks for finding the fix that unblocked us @EdAllonby!

@petyosi We just ran into this issue as well on IE 11. For me, simply put, `ag-grid-react` (v19) straight out did not work when it was included in our vendor bundle.

Our specific issue was:

> unable to get property 'AddEventListener' of undefined or null reference

We got this with the most basic possible usage of the grid:

**Webpack config**
```javascript
    entry: {
        main: ['babel-polyfill', path.join(__dirname, '../src/app/index')],
        vendor: ['ag-grid-community', 'ag-grid-react', 'react', 'react-dom', 'react-router'],
    },
```

**SomeComponent**
```javascript
<AgGridReact columnDefs={[]} rowData={[]} />
```

The issue was immediately resolved by removing 'ag-grid-community' and 'ag-grid-react' from the vendor bundle.

This is clearly an issue of something in the grid not working in IE 11.

As to why removing the grid packages from the webpack config fixed the problem, my best guess - not having a deep understanding of  how `require`/`import` work once on the browser, is that when included in the vendor bundle, if the vendor bundle loads before the main bundle, the grid or the `require` wrapping code around the grid somehow captures some browser references in a closure, which would explain why the errors occur in user-land even after we have loaded the polyfill and our code. By taking the packages out of the vendor bundle, we ensure that the polyfill is run before the grid code is ever executed/wrapped on the browser.
